api/v1/cluster_webhook.go

```diff
@@ -1431,26 +1440,46 @@ func (r *Cluster) validateMinSyncReplicas() field.ErrorList {
 }
 
 func (r *Cluster) validateStorageSize() field.ErrorList {
-	return validateStorageConfigurationSize("Storage", r.Spec.StorageConfiguration)
+	return validateStorageConfigurationSize(*field.NewPath("Storage"), r.Spec.StorageConfiguration)
 }
 
 func (r *Cluster) validateWalStorageSize() field.ErrorList {
 	var result field.ErrorList
 
 	if r.ShouldCreateWalArchiveVolume() {
-		result = append(result, validateStorageConfigurationSize("walStorage", *r.Spec.WalStorage)...)
+		result = append(result,
+			validateStorageConfigurationSize(*field.NewPath("walStorage"), *r.Spec.WalStorage)...)
+	}
+
+	return result
+}
+
+func (r *Cluster) validateTablespacesStorageSize() field.ErrorList {
+	if r.Spec.Tablespaces == nil {
+		return nil
 	}
 
+	var result field.ErrorList
+
+	for tablespaceName, tablespaceConf := range r.Spec.Tablespaces {
+		result = append(result,
+			validateStorageConfigurationSize(
+				*field.NewPath("tablespaces", tablespaceName), tablespaceConf.Storage)...,
+		)
+	}
 	return result
 }
 
-func validateStorageConfigurationSize(structPath string, storageConfiguration StorageConfiguration) field.ErrorList {
+func validateStorageConfigurationSize(
+	structPath field.Path,
+	storageConfiguration StorageConfiguration,
+) field.ErrorList {
 	var result field.ErrorList
 
 	if storageConfiguration.Size != "" {
 		if _, err := resource.ParseQuantity(storageConfiguration.Size); err != nil {
 			result = append(result, field.Invalid(
-				field.NewPath("spec", structPath, "size"),
+				field.NewPath("spec", structPath.String(), "size"),
```

Comment:
Since `spec` should be already in the `structPath` field, this is really `structPath.Child("size")`